### ISSUE TITLE
rustc interrupted by SIGSEGV

### ISSUE THREAD

**Issue by arturamiryan** [rust-lang/rust#123025]
I tried this code on Armbian on MangoPi MQ-Pro, RISC-V:

```rust
fn main() {
    println!("Hello, world!");
}
```

I expected to see this happen: standard compilation

Instead, this happened: 
```
error: rustc interrupted by SIGSEGV, printing backtrace

/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0xbe20c8)[0x3f995490c8]
linux-vdso.so.1(__vdso_rt_sigreturn+0x0)[0x3fa0e92800]
/lib/riscv64-linux-gnu/libc.so.6(read+0x48)[0x3f987d390a]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/libstd-1c886dc89bd84c1d.so(_ZN3std3sys3pal4unix2fs4File4read17hab68df65f645bf60E+0x26)[0x3f988d741c]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0x53fd826)[0x3f9dd64826]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0x53ff202)[0x3f9dd66202]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0x53ff776)[0x3f9dd66776]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0x53ff7c6)[0x3f9dd667c6]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0x53ffe8e)[0x3f9dd66e8e]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-63b44ee064357158.so(+0x5400a50)[0x3f9dd67a50]
/home/user/.rustup/toolchains/stable-riscv64gc-unknown-linux-gnu/bin/../lib/libstd-1c886dc89bd84c1d.so(rust_metadata_std_7a07c3436a7b4eae+0x7ad72)[0x3f988dcd72]
/lib/riscv64-linux-gnu/libc.so.6(+0x6955a)[0x3f9878f55a]
```

### Meta

`rustc --version --verbose`:
```
rustc 1.79.0-nightly (0824b300e 2024-03-24)
binary: rustc
commit-hash: 0824b300eb0dae5d9ed59719d3f2732016683d66
commit-date: 2024-03-24
host: riscv64gc-unknown-linux-gnu
release: 1.79.0-nightly
LLVM version: 18.1.2
```

<details><summary>Backtrace</summary>
<p>

```
error: rustc interrupted by SIGSEGV, printing backtrace

/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-909356f78f49b3cc.so(+0xc5ef12)[0x3fb6604f12]
linux-vdso.so.1(__vdso_rt_sigreturn+0x0)[0x3fbdfce800]
/lib/riscv64-linux-gnu/libc.so.6(read+0x48)[0x3fb57fd90a]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/libstd-9cb9eca00d5d78f0.so(_ZN3std3sys3pal4unix2fs4File4read17h8a74c485596d051fE+0x20)[0x3fb59086ca]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-909356f78f49b3cc.so(+0x587353a)[0x3fbb21953a]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-909356f78f49b3cc.so(+0x5873b26)[0x3fbb219b26]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-909356f78f49b3cc.so(+0x5873b76)[0x3fbb219b76]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-909356f78f49b3cc.so(+0x5874022)[0x3fbb21a022]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/librustc_driver-909356f78f49b3cc.so(+0x5874f26)[0x3fbb21af26]
/home/user/.rustup/toolchains/nightly-riscv64gc-unknown-linux-gnu/bin/../lib/libstd-9cb9eca00d5d78f0.so(rust_metadata_std_85a3b2a8897af422+0x82ec6)[0x3fb590eec6]
/lib/riscv64-linux-gnu/libc.so.6(+0x6955a)[0x3fb57b955a]

note: we would appreciate a report at https://github.com/rust-lang/rust
help: you can increase rustc's stack size by setting RUST_MIN_STACK=16777216
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.27s
```

</p>
</details>

**Comment by saethlin** [comment]
This looks an awful lot like https://github.com/rust-lang/rust/issues/117022

**Comment by bjorn3** [comment]
Which kernel version do you have? If this is indeed the same issue as #117022 there was a regression which is fixed in linux 6.6 according to https://github.com/rust-lang/rust/issues/117022#issuecomment-1841570340.

**Comment by arturamiryan** [comment]
> Which kernel version do you have? (`uname -a` will show this) If this is indeed the same issue as #117022 there was a regression which is fixed in linux 6.6 according to [#117022 (comment)](https://github.com/rust-lang/rust/issues/117022#issuecomment-1841570340).

Yeah, that is probably it. Switched kernel version and it works again. Closing the issue, thanks for the help!
